hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java

```diff
@@ -437,4 +473,36 @@ private static ServiceException createLeaderNotReadyException(
     return new ServiceException(leaderNotReadyException);
   }
 
+  private static OmBucketInfo getOmBucketInfo(OzoneManager ozoneManager,
+      OmBucketInfo buckInfo, String volName, String buckName) {
+    String buckKey =
+        ozoneManager.getMetadataManager().getBucketKey(volName, buckName);
+    try {
+      buckInfo =
+          ozoneManager.getMetadataManager().getBucketTable().get(buckKey);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return buckInfo;
+  }
+
+  private static BucketLayout getBucketLayout(OmBucketInfo buckInfo) {
+    if (buckInfo != null) {
+      return buckInfo.getBucketLayout();
+    } else {
+      buckInfo = null;
+      // TODO: Handle bucket validation
```

Comment:
bucket null is a tricky case. Since this bucketTable is not inside BUCKET_LOCK the consistency won't be guaranteed. I think, if the bucket is coming as null then how about acquiring a BUCK_LOCK and re-read the bucket key again?

Reply:
@aryangupta1998 Since `#validateBucketAndVolume()` check is there in every OMClientRequest , how about to return `BucketLayout.LEGACY` in case of a null bucket. For debugging purpose, please add warn log message saying that bucket doesn't exists.

`LOG.error("Bucket not found: {}/{} ", volumeName, bucketName);`

Reply:
Done

Reply:
Thanks @aryangupta1998 for addressing this. I didn't think about a unit test case verifying the default `BucketLayout.LEGACY` path in my previous comment. Can we add a new test case where the bucket doesn't exists and verify it. Probably you can add a simple test in TestRootedOzoneFileSystem or TestRootedOzoneFileSystemWithFSO.

Verify creating a file or key on a non-existent bucket something like,

`bucketPath = new Path(volumePath, "invalidBucket");`

Reply:
@aryangupta1998 Please refer below unit test case and can use this kinda test to verify our code path. Since all the RPC key calls are invoked on bucket, a mock based unit test will work.
```
  @Test(timeout = 300_000)
  public void testRequestWithNonExistentBucket() throws Exception {
    ozoneManager = Mockito.mock(OzoneManager.class);
    ozoneConfiguration.set(OMConfigKeys.OZONE_OM_DB_DIRS,
        folder.newFolder().getAbsolutePath());
    omMetadataManager = new OmMetadataManagerImpl(ozoneConfiguration);
    when(ozoneManager.getMetadataManager()).thenReturn(omMetadataManager);

    String volumeName = "vol1";
    String bucketName = "invalidBuck";
    OzoneManagerProtocolProtos.OMRequest omRequest =
        TestOMRequestUtils.createCompleteMPURequest(volumeName, bucketName,
            "mpuKey", "mpuKeyID", new ArrayList<>());

    OMClientRequest req = OzoneManagerRatisUtils.createClientRequest(omRequest, ozoneManager);
    Assert.assertNotNull(req);
    Assert.assertTrue("Unexpected request on invalid bucket", req instanceof S3MultipartUploadCompleteRequest);
  }
```

Reply:
Done